db/migrate/20150817163600_deduplicate_user_identities.rb

```diff
@@ -1,7 +1,7 @@
 class DeduplicateUserIdentities < ActiveRecord::Migration
   def change
     execute 'DROP TABLE IF EXISTS tt_migration_DeduplicateUserIdentities;'
-    execute 'CREATE TEMPORARY TABLE tt_migration_DeduplicateUserIdentities AS SELECT id,provider,user_id FROM identities;'
```

Comment:
@stanhu I have since made a rebased merge request [on gitlab](https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/1366). 

However I am getting strange build errors: 

```Progress: |WARNING: This version of Gitlab depends on gitlab-shell 2.6.5, but you are running on Unknown. Please update gitlab-shell.``` -> That's seems a problem of a outdated gitlab-shell on the build machine.

<img width="1280" alt="screen shot 2015-09-21 at 10 56 10 am" src="https://cloud.githubusercontent.com/assets/5023034/9997164/5c4bb46e-6093-11e5-80e4-7f79844cc08b.png">
<img width="1280" alt="screen shot 2015-09-21 at 10 56 14 am" src="https://cloud.githubusercontent.com/assets/5023034/9997176/6cf2f94e-6093-11e5-8e74-89771a65ea0d.png">
<img width="1280" alt="screen shot 2015-09-21 at 10 55 48 am" src="https://cloud.githubusercontent.com/assets/5023034/9997189/84514992-6093-11e5-93ac-650d161484a2.png">

On the gitlab merge request I also refactored the code to take note of @dosire's scalability concern and uploaded screenshots of the changes as stated on the contribution guidelines.

I have restarted the builds to see if they will pass.

PS: I felt that gitlab would be a better place to push a merge request but I should have been prudent enough to cancel this pull request since.